src/tree.c

```diff
@@ -111,6 +111,7 @@ void ofi_rbmap_cleanup(struct ofi_rbmap *map)
 void ofi_rbmap_destroy(struct ofi_rbmap *map)
 {
 	ofi_rbmap_cleanup(map);
+	free(map->free_list);
```

Comment:
This isn't quite right.  free_list is a singly linked list of ofi_rbnode entries, linked together using the node->right pointer.  The memory leak is a result of not freeing those entries.  In your specific case, there's only 1 entry, but there could be more.  We need to iterate over the free_list and free all entries.  E.g.
```
while (map->free_list) {
    node = map->free_list;
    map->free_list = node->right;
    free(node);
}
```

Reply:
It looks like we would need to cherry-pick this fix into v1.9.x as well.